### PR TITLE
fix(component-testing): correct imports for relative paths in cypress.json 

### LINKED ISSUE #16053: Component Tests Can't Run When Cypress Is In a Different Project Folder
### Current behavior
I have a cypress project where we have the cypress config in a different folder, so that we run `cypress open --project spec/e2e`.  This has always worked fine with the traditional cypress e2e tests.

I tried out the component tests and encountered a bug where the cypress test runner doesn't seem able to execute the test.  (screenshot below).  The webpack compiles successfully, the tests are shown correctly, but clicking on a test results in the console error below 👇 

Screen Shot 2021-04-18 at 10 25 49 AM

If I move the cypress config to the root folder, as is the more common case with cypress examples/docs, everything works fine.  It appears to be a path resolution error.

### Desired behavior
Able to run component tests inside of a different project folder.

### Test code to reproduce
**Minimal Repo to observe, with additional instructions in the readme:  https://github.com/peterkrieg/cypress-bug-repro**

Thank you for looking at this!  Cypress component testing looks awesome and I can't wait to try it some more.

### Versions

`Cypress`:  7.1.0
`@cypress/webpack-dev-server`: 1.1.3

### PR DESCRIPTION
**Pull request by lmiller1990**
- Closes #16053

### User facing changelog

Use absolute paths instead of relative paths for webpack dev server and when requesting specs from the CT runner.

### Additional details

This solves the problem in #16053 where having specs in a relative directory above a location specified with `--project` are not loaded. Now you may specify a relative directory for `componentFolder`.

I added an example project that reproduces this problem in `server-ct/test....`. This does not run on CI yet, but the goal will be something like `packages/server`, where all test projects go in `cypress/fixtures/projects` and we can run them using something like `e2e.it`. 

### How has the user experience changed?

Supports specifying a relative `componentFolder`.

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has the original issue or this PR been tagged with a release in ZenHub? 
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: npm/webpack-dev-server/src/loader.ts [hunk 1]
```diff
@@ -15,8 +17,8 @@ const makeImport = (file: Cypress.Cypress['spec'], filename: string, chunkName:
   const magicComments = chunkName ? `/* webpackChunkName: "${chunkName}" */` : ''
 
   return `"${filename}": {
-    shouldLoad: () => document.location.pathname.includes(${JSON.stringify(file.relative)}),
-    load: () => import(${JSON.stringify(path.resolve(projectRoot, file.relative))} ${magicComments}),
+    shouldLoad: () => document.location.pathname.includes(${JSON.stringify(file.absolute)}),
+    load: () => import(${JSON.stringify(path.resolve(projectRoot, file.absolute))} ${magicComments}),
```
Comment (agg23) [inline npm/webpack-dev-server/src/loader.ts:21]: `path.resolve` does not need the `projectRoot`, as the file path is absolute.

File: packages/runner-ct/src/app/RunnerCt.tsx [hunk 2]
```diff
@@ -92,7 +92,9 @@ const RunnerCt = namedObserver('RunnerCt',
 
     const runSpec = React.useCallback((file: FileNode) => {
       setActiveIndex(0)
-      const selectedSpec = props.state.specs.find((spec) => spec.absolute.includes(file.relative))
+      // We request an absolute path from the dev server but the spec list displays relative paths
+      // For this reason to match the spec we remove leading relative paths. Eg ../../foo.js -> foo.js.
+      const selectedSpec = props.state.specs.find((spec) => spec.absolute.includes(file.relative.replace(/\.\.\//gi, '')))
```
Comment (agg23) [inline packages/runner-ct/src/app/RunnerCt.tsx:97]: This regex `replace` is called on every item in the loop. Please assign it to a variable instead.

**Review by agg23 — CHANGES_REQUESTED** [review]

File: npm/webpack-dev-server/src/loader.ts (hunk 1 above)
Comment (lmiller1990) [reply npm/webpack-dev-server/src/loader.ts]: Fixed.

File: packages/runner-ct/src/app/RunnerCt.tsx (hunk 2 above)
Comment (agg23) [reply packages/runner-ct/src/app/RunnerCt.tsx]: I mean:
```
const filePath = file.relative.replace(/\.\.\//gi, '')
const selectedSpec = props.state.specs.find((spec) => spec.absolute.includes(filePath))
```

**Review by agg23 — DISMISSED** [review]

File: packages/runner-ct/src/app/RunnerCt.tsx (hunk 2 above)
Comment (lmiller1990) [reply packages/runner-ct/src/app/RunnerCt.tsx]: 🤦‍♂️ right.

Done.

**Review by agg23 — APPROVED** [review]

**Pull request merged**